### ISSUE TITLE
How to update the correct speed for My Hashrate?

### ISSUE THREAD

**Issue by vanthoi** [MPOS/php-mpos#2751]
Hi developer. I'm using stratum z-nomp server & running with Equihash algorithm. I tried to find out before asking the question. But I have not found a solution in all the answers that the development team has left on google. I use the mpos version for my z-nomp server here: https://github.com/xiaolin1579/z-mpos. Since the latest version of MPOS I can not fix the My Sharerate error. I tried to run as many miners. But it still can not update My Sharerate. So I used the old version of xiaolin1579. Although the new version of the interface has been optimized and a lot better. As for My Hashrate, in any case, I have encountered an incorrect hash speed error. For example, my miners have 1700 Sol / s, My Hashrate is 3200 Sol / s. Is there any way for me to update the speed accurately for My Hashrate? One time I would like to thank the development team MPOS for giving the web a good webpotal.

**Comment by r4sas** [comment]
See here: https://github.com/MPOS/php-mpos/commit/30668ed502c3fe948c9a951313e0c101daa532ef

**Comment by vanthoi** [comment]
@r4sas As I mentioned above. The My Hashrate section of my dashboard page is incorrect. For the MPOS version 1.1 I have installed and have checked it is still incorrect. In this new version, the My Sharerate section is not up to date. So I went back to using the old version of @ xiaolin1579. I tried updating the code that you changed above into the old version. But everything is still no change. I try running cron again, reopening server z-nomp. But still does not solve the problem!

**Comment by vanthoi** [comment]
For easier visualization. I send two more images below. People can see the hash speed is not as accurate as the speed of the miners. Even at acceptable tolerances.
![image](https://user-images.githubusercontent.com/13189483/42420854-7b47d57c-82f6-11e8-885b-37a098417479.png)
![image](https://user-images.githubusercontent.com/13189483/42420858-8f7c9456-82f6-11e8-84f8-546d739be23f.png)

**Comment by TheSerapher** [comment]
I am not sure but could this be cause by a wrong coin class or stratum inserting shares at a different diff into the database? 

I fail to see a real issue with MPOS here but rather some specific thing for this coin that needs adjustment.

**Comment by vanthoi** [comment]
@TheSerapher First I thank you for answering my question. Also, thanks to my friend, I fixed a lot of bug related to MPOS on google. Well I'm sure it's not a matter of MPOS. I just can not find a solution. So I've got the help of the developer to see where I need to customize it in MPOS or I have to customize Z-NOMP's VARDIFF. Although the speed of the hash in DASHBOARD is not accurate, But things still work so well until now. Just that I do not want people to come to my pool to tell me that the pool has fake their hash speed. Then they leave again. I want someone to point me in the direction that I can find out and fix it.

**Comment by r4sas** [comment]
Have you set $target_diff for your coin profile?
https://github.com/xiaolin1579/z-mpos/blob/master/include/classes/coins/coin_equihash.class.php

**Comment by vanthoi** [comment]
@r4sas my profile coin:
`"ports": {
        "3334": {
            "tls":false,
	    "diff": 0.05,
	    "varDiff": {
                "minDiff": 0.04,
                "maxDiff": 24,
                "targetTime": 60,
                "retargetTime": 80,
                "variancePercent": 30
            }
        }
    }`
Are you saying this?

**Comment by r4sas** [comment]
@vanthoi no, I'm about profile in `include/classes/coins/`.

**Comment by vanthoi** [comment]
@r4sas Đối với file coin_equihash.class.php nó đã có sẵn trong phiên bản của  @xiaolin1579

**Comment by TheSerapher** [comment]
Please stick with english, threads are not useful if you change language. Hard to follow them for others reading them later on. :-)

**Comment by vanthoi** [comment]
@r4sas  Can you give me any suggestion so that I can update the hash speed correctly for My Hashrate ?

**Comment by r4sas** [comment]
@vanthoi try edit `$target_bits`.

How hashrate calculated:
https://github.com/MPOS/php-mpos/blob/236cccd5d3178e99504bf2665db8bddfed6aa176/include/classes/coins/coin_base.class.php#L50-L56

**Comment by vanthoi** [comment]
Sorry @r4sas . Do you update your answer on https://github.com/MPOS/php-mpos/issues/2751#issuecomment-403293733 that I did not look back. I have updated the difficulty in my configuration of 256. But I do not see the change.

**Comment by r4sas** [comment]
@vanthoi because you using different pool software, that value (in global.inc.php) not needed. it used only for pushpoold stratum software. 

Please read FAQ: https://github.com/MPOS/php-mpos/wiki/FAQ#why-does-stratum-seem-to-be-off-when-using-mpos

**Comment by vanthoi** [comment]
@r4sas Well sorry you. I was troubled by repeatedly fixing that was not resolved. I've been trying to find out for a  week now. So now I need to change something in MPOS so that it can be correct.

**Comment by TheSerapher** [comment]
@vanthoi since there is nothing wrong in MPOS I suggest you ask around other pool operators, forums or reddit to find a solution to your issue. The only setting that would affect your coins hash rate in MPOS is the mentioned `$target_bits` value mentioned by @r4sas already. I am aware that pool operators like to sit on their knowledge and not share it to maximise profits while minimising their competition, but hopefully someone will share it with you.

Closing since this is a coin specific issue you'd have to bring up with coin devs or pool operators for that coin.

**Comment by vanthoi** [comment]
Although support was closed but I tried to customize it to achieve the fastest results. One time I thank you for your support. I wish you a nice day!
